src/main/sensors/gyro_filter_impl.h

```diff
@@ -11,7 +11,15 @@ static FAST_CODE void GYRO_FILTER_FUNCTION_NAME(gyroSensor_t *gyroSensor, timeDe
         if (isDynamicFilterActive()) {
             if (axis == X) {
                 GYRO_FILTER_DEBUG_SET(DEBUG_FFT, 0, lrintf(gyroADCf)); // store raw data
-                GYRO_FILTER_DEBUG_SET(DEBUG_FFT_FREQ, 3, lrintf(gyroADCf)); // store raw data
+                GYRO_FILTER_DEBUG_SET(DEBUG_FFT_FREQ, 3, lrintf(gyroADCf)); // store raw data in 3
+            }
+            if (gyroConfig()->dyn_filter_location == DYN_FILTER_BEFORE_STATIC_FILTERS) {
```

Comment:
@DieHertz: Can you have a look to check if this is not having a negative impact with respect to the optimisation?

Reply:
I will look at the assembly tonight, it has a huge chance to do so.

Reply:
If the option to use PT1 is potentially problematic, I think we can drop back to just Notch.  Our testing has strongly shown notch to be superior.  The old problem was that the notch wasn't working all that well, because it wasn't centering on the noise peak.  Not it sits well on the noise in the center of the peak, so the Notch seems preferable.

My gut feeling is that we should take the PT1 option out, which would greatly simplify the code etc

Reply:
Location is implemented wrong way. The only part of the code that should move if `gyroDataAnalysePush`, not the entire block.

Reply:
Presence or absence of PT1 doesn't change this specific part of code in any way.

Reply:
originally i had just one #ifdef USE_GYRO_DATA_ANALYSE block, and moved the fixed notches before and after it, but I guess that has the same problem

Reply:
Only thing that should be moved is `gyroDataAnalysePush` call, not the actual application of filter.